### ISSUE TITLE
Build gh-pages for Amber 0.14.3

### ISSUE THREAD

**Issue by tomrake** [amber-smalltalk/trysmalltalk#4]
This is dependent on PR #3 and once done will allow #2 to be closed.

**Comment by hhzl** [comment]
Now ready to be done.

**Comment by tomrake** [comment]
Please review, I think this is fine.

My website image thread (gh-pages) is: https://github.com/tomrake/trysmalltalk/tree/gh-pages

You can see the example at:  https://tomrake.github.io/trysmalltalk/

While a PR is possible for this it will contain all the changes in bower_components.

If this is fine I will force push this to amber-smalltalk/trysmalltalk gh-pages and close this issue.

**Comment by hhzl** [comment]
I think think this is fine. There is no way around the changes in bower_components in the gh-branch.

Please force-push and close.

**Comment by mkroehnert** [comment]
force-pushing into the main repository is not a good idea and should not be done.

@tomrake what is the exact problem with opening a pull request to the gh-branch of this repo?

**Comment by herby** [comment]
Force-pushing into feature branch is not a problem, but gh-pages is sort-of master for the view-side of the repo. I would also like to know what is the problem. If it is the big amount of changes in bower_components (which must be pre-filled for gh-pages), this is not a problem. There are even schools of thought who think things like node_modules should always be pushed into git repo (http://stackoverflow.com/questions/11459475/should-i-check-in-node-modules-to-git-when-creating-a-node-js-app-on-heroku) though they seem to be in decline.

**Comment by hhzl** [comment]
I do not see the whole issue as a problem. It is just about updating to 0.14.1.

Yes, gh-pages may be considered master for the view-side of the repo. 

Maybe we should have a 0.14.1 branch to work on this (later to be merged into gh-pages).

But surely we need a branch 0.14.2 to work on  #15

**Comment by mkroehnert** [comment]
>
> I do not see the whole issue as a problem. It is just about updating to
> 0.14.1.
>
If you don't see this as a problem you should never be recommending doing
force-pushes.
Force-pushes to public master repositories is one of the worst thing you
can do.
Your own forks/clones are another story and what you do there is up to you.

Since this issue is just about updating, there is no real problem in doing
a regular commit and providing a pull request to the correct branch.

**Comment by mkroehnert** [comment]
+1 for creating version branches which are then merged into the gh-pages
branch.

**Comment by hhzl** [comment]
I understand that @tomrake wants to force-push to the gh-pages branch. This branch is need for the project to be displayed properly at http://amber-smalltalk.github.io/trysmalltalk.

    https://tomrake.github.io/trysmalltalk/

is fine

    Smalltalk version '0.14.1' 

versus
     
     http://amber-smalltalk.github.io/trysmalltalk/

     Smalltalk version '0.14.1'

**Comment by tomrake** [comment]
Gentlemen, the problem here as I see that versioning the website dependencies bower_components and node_modules are completely rebuildable by the process `npm install; bower install;grunt devel` commands the only reason the bower_components and node_modules are included is because the gh-pages branch is a website image branch. 

When including the bower_components your have to trust that `git add -A` works correctly and that you have not added extra cruft to be versioned.

After doing several upgrade cycles procuding 0.14.0,0.14.1,0.14.3 I am getting systemic process on building the gh-pages branches. I am going to close building a 0.14.1 page in favor of a 0.14.3 version and  try this again. The 0.14.3 version is a fast foward merge.

**Comment by tomrake** [comment]
Again here is what propose, the final version will only be build with a small change to the bower.json  where points trysmalltalk to the amber-smalltalk based repo instead of the tomrake based repo.

#18 Need to be merged before this can happen.
 
This is what is looks like from my repo.

https://tomrake.github.io/

I think both  #18 and the gh-pages merger will be non forced.

**Comment by hhzl** [comment]
http://amber-smalltalk.github.io/trysmalltalk/

has
      
         Smalltalk version '0.13.3'
